bloom/bloom.go

```diff
@@ -4,13 +4,15 @@ import (
 	// "fmt"
 
 	"github.com/willf/bitset"
+	"sync"
 )
 
 const defaultSize = 2 << 24
 
 var seeds = []uint{7, 11, 13, 31, 37, 61}
 
 type BloomFilter struct {
+	mu    sync.Mutex
```

Comment:
this can use RWMutex, then `bf.Contains` can use `bf.mu.RLock()`

Reply:
I ran some benchmarks, and these are the results:

1. Current implementation (`sync.Mutex`):
```
goos: linux
goarch: amd64
pkg: github.com/penglongli/gin-metrics/ginmetrics
cpu: Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
BenchmarkMiddlewareSamePort-4   	   10000	    187130 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    368713 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    376937 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    360904 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    178943 ns/op
PASS
ok  	github.com/penglongli/gin-metrics/ginmetrics	19.239s
```
2. Suggesteg implementation (`sync.RWMutex`, where `bf.Contains()` is protected by `bf.mu.RLock()`):
```
goos: linux
goarch: amd64
pkg: github.com/penglongli/gin-metrics/ginmetrics
cpu: Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
BenchmarkMiddlewareSamePort-4   	   10000	    179478 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    364757 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    365689 ns/op
BenchmarkMiddlewareSamePort-4   	   10000	    180693 ns/op
BenchmarkMiddlewareSamePort-4   	    9828	    360628 ns/op
PASS
ok  	github.com/penglongli/gin-metrics/ginmetrics	18.937s
```

Benchtest comparison:
```
name                  old time/op  new time/op  delta
MiddlewareSamePort-4   295µs ±39%   290µs ±38%   ~     (p=0.690 n=5+5)
```
 As it can be seen the results are statistically indistinguishable. Increased the number of times benchmarks were run up to 25, and the result is still the same.

```
name                  old time/op  new time/op  delta
MiddlewareSamePort-4   278µs ±38%   264µs ±45%   ~     (p=0.658 n=25+25)
```

I suggest to keep it the way it is.